### DISCUSSION TITLE
Form Autocomplete Feature

### DISCUSSION THREAD

**Discussion by chunleng** [Jelmerro/Vieb#457] [Ideas]
I have been thinking of implementing something for my daily use and probably benefit users using vieb browser.
Not sure if this is the right vision from Jelmerro, but I imagine myself as the typical user of vieb. Someone who is a heavy vim user and likes to add their own scripts and plugin to help automate the browsing.

Therefore, the autocomplete feature should integrate the developer mindset in the way that it can allow high customizability without much hassle.

---

## What I propose

A customizable form element filling function

### How to config

In a general direction, user provide config such as

```
[
  {
    "target": [{"match_regex": ".*name.*", "html_attr": ["name","id"]}],
    "target_join": "all",
    "perform": {"type": "fill", "metadata": {"value": "Jelmerro"}}
  },
  {
    ...,
    "perform": {"type": "password", "metadata": {"type": "lastpass", "key": "${site_domain}"}}
  },
]
```

The password type is an illustration on the flexibility of the autocomplete system as such we can use this system to extend the function of other completion means

**Comment by Jelmerro** [comment]
There is definitely a need for an auto-fill feature in Vieb, but I'm still undecided on the storage format/structure. I would like to avoid a huge range of JSON configs that would need to be used for it to work.
